python/ray/dag/tests/experimental/test_torch_tensor_dag.py

```diff
@@ -1792,6 +1796,84 @@ def test_torch_tensor_nccl_all_reduce_scheduling_one_ready_group(ray_start_regul
     compiled_dag.teardown()
 
 
+@pytest.mark.parametrize("ray_start_regular", [{"num_cpus": 4}], indirect=True)
+def test_torch_tensor_copy_before_nccl_all_reduce(ray_start_regular):
+    """
+    Test that a tensor is copied before all reduce is launched.
+    That is, the tensor is not all-reduced in place and can be reused
+    in the DAG as input to other nodes.
```

Comment:
Nit: "as inputs to other nodes in the DAG."

Reply:
ack